genrl/deep/common/base_class.py

```diff
@@ -0,0 +1,139 @@
+from abc import ABC
+from typing import Any, Dict, Optional, Tuple, Union
+
+import numpy as np
+import torch
+
+from genrl.deep.common.utils import load_params, save_params, set_seeds
+
+
+class BaseAgent(ABC):
+    def __init__(self, network_type: str, env: Any, epochs: int = 100, **kwargs):
+        self.network_type = network_type
+        self.env = env
+        self.epochs = epochs
+        self.seed = kwargs.get("seed", None)
+        self.render = kwargs.get("render", False)
+        self.run_num = kwargs.get("run_num", None)
+        self.save_model = kwargs.get("save_model", None)
+        self.load_model = kwargs.get("load_model", None)
+        self.save_interval = kwargs.get("save_interval", 50)
```

Comment:
Are you setting this way for a particular reason?

Reply:
Yes this sets a default value to the variables. So if we are using it somewhere in the code unknowingly to it doesn't throw errors.

Reply:
Wouldn't that happen automatically if we specify the default value in the function definition?

Reply:
yes, but this was to  be on the safer side

Reply:
won't really make a big difference tbh

Reply:
Right, just saying because its not a standard way to do it